styhead/rz sbc: support audio, ov5640 and dsi panel

tiennguyenuh opened the pull request:
Since these patches are in review in the meta-renesas repository in this PR for Weston [PR #33](https://github.com/Renesas-SST/meta-renesas/pull/33) for Linux recipe, this PR aims to create a dedicated branch in the Linux repository for better management and tracking.

This PR includes all the development work completed so far.
- Fixed GPU soft reset trigger when using the Panfrost driver.
- Added audio support for RZ/G2L-SBC.
- Implemented a workaround to retry CRU initialization when there is no input from the MIPI CSI2 device.
- Added WDT2 clock and reset support for the R9A07G044 SoC.
- Added Waveshare panel support for RZ/G2L-SBC.

tiennguyenuh committed:
media: rzg2l-cru: add WA to retry CRU initialization when there is no input from MIPI CSI2 device

Add a WA to retry CRU initialization when there is no
input from MIPI CSI2 device.

This workaround creates a thread to monitor the CRU status. If the CRU
is not in a running state, the thread will stop the CRU and
reinitialize the processing.

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
arm64: dts: rzpi: restore power-domains property

Restores the `power-domains` property to the GPU node when
using panfrost driver to prevent triggerring a GPU soft reset.

[12.643867] panfrost 11840000.gpu: gpu sched timeout, js=0, config=0x7300, status=0x8, head=0xa0045c0, tail=0xa0045c0, sched_job=00000000175dd3e3
[12.672055] panfrost 11840000.gpu: gpu soft reset timed out
[12.682527] ------------[ cut here ]------------
[12.691992] WARNING: CPU: 0 PID: 92 at drivers/gpu/drm/panfrost/panfrost_job.c:164 panfrost_job_run+0x2f4/0x310

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
clk: renesas: add WDT2 clocks and reset support for r9a07g044

This patch adds support for two new watchdog timer (WDT2) clocks and their corresponding
reset entries for the R9A07G044 SoC.

- Adds `wdt2_pclk` and `wdt2_clk` to the `common` clock array.
- Adds `R9A07G044_WDT2_PRESETN` to the reset array for WDT2 reset handling.

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
arm64: rzpi: support audio for RZ/G2L-SBC

Changes include:
    - Add audio configuration in rzpi.dts.
    - Change the interrupt trigger to falling edge instead of level low,
      as the Renesas interrupt controller does not support clearing the
      low-level detection interrupt, making it unsuitable
      for use in the driver.

Fixes the following error observed during boot:
[1.751706] genirq: Setting trigger mode 8 for irq 61 failed (rzg2l_gpio_irq_set_type+0x0/0x40)
[1.770223] da7219 0-001a: Failed to request IRQ: -22

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

long-trinh-rvc committed:
drivers: rcar-du: rzg2l: add clock calculation logic for DSI in DU timing driver

This commit adds clock calculation logic for DSI within the DU
timing driver, ensuring proper handling of clock settings based on DSI
status.

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

vudangRVC commented:
Make sure you guys have checked the patches by using `scripts/checkpatch.pl`.
Refer to: [Posting patches](https://docs.kernel.org/process/5.Posting.html)

I'll look closer into the code when the final commit of MIPI DSI support is pushed.

tiennguyenuh committed:
drivers: gpu: panel: add waveshare panel support for RZG2L SBC board

Added support for Waveshare 5 inch Touchscreen Monitor MIPI-DSI LCD on
the RZG2L SBC board.

Upstream-Status: Pending

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
arm64: defconfig: Add rzpi_defconfig

This commit adds a defconfig for RZ/G2L-SBC, based on the
renesas_defconfig from tag renesas-defconfig-v6.6 on
git://git.kernel.org/pub/scm/linux/kernel/git/geert/renesas-devel.git

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

preetam-reddy commented:
The same patches have been reviewed and merges in meta-reneas pull request: https://github.com/Renesas-SST/meta-renesas/pull/33
Hence skipping review and merging.
